File: internal/testdata/release-smoke/spring-vuln/Controller.java
// Spring vulnerable fixture: string-concatenated SQL query.
// PF-SPRING-SQLI-004 should fire on the concatenation in the @GetMapping method.
import org.springframework.web.bind.annotation.*;
import org.springframework.jdbc.core.JdbcTemplate;

@RestController
@RequestMapping("/api")
public class UserController {

    private final JdbcTemplate jdbc;

    public UserController(JdbcTemplate jdbc) {
        this.jdbc = jdbc;
    }

    @GetMapping("/users")
    public String search(@RequestParam String name) {
        // VULNERABLE: string concatenation in SQL query
        String sql = "SELECT name FROM users WHERE name = '" + name + "'";
        return jdbc.queryForObject(sql, String.class);
    }

    @GetMapping("/redirect")
    public String redirect(@RequestParam String url) {
        // VULNERABLE: open redirect
        return "redirect:" + url;
    }
}

File: internal/testdata/release-smoke/laravel-vuln/Controller.php
<?php
// Laravel vulnerable fixture: raw SQL with string concatenation.
// PF-LARAVEL-SQLI-002 should fire on the DB::raw with concatenation.

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class UserController extends Controller
{
    public function search(Request $request)
    {
        $name = $request->get('name');
        // VULNERABLE: raw SQL with string concatenation
        $users = \DB::select('SELECT * FROM users WHERE name = \'' . $name . '\'');
        return response()->json($users);
    }

    public function redirect(Request $request)
    {
        $url = $request->input('url');
        // VULNERABLE: open redirect
        return redirect()->away($url);
    }
}
